packages/output/src/browser/output-contribution.ts

```diff
@@ -122,19 +122,29 @@ export class OutputContribution extends AbstractViewContribution<OutputWidget> i
     registerCommands(registry: CommandRegistry): void {
         super.registerCommands(registry);
         registry.registerCommand(OutputCommands.CLEAR__WIDGET, {
-            isEnabled: widget => this.withWidget(widget),
-            isVisible: widget => this.withWidget(widget),
-            execute: () => this.widget.then(widget => widget.clear())
+            isEnabled: arg => {
+                if (arg instanceof Widget) {
+                    return arg instanceof OutputWidget;
+                }
+                return this.shell.currentWidget instanceof OutputWidget;
+            },
+            isVisible: arg => {
+                if (arg instanceof Widget) {
+                    return arg instanceof OutputWidget;
+                }
+                return this.shell.currentWidget instanceof OutputWidget;
+            },
+            execute: () => this.tryGetWidget()?.clear()
         });
         registry.registerCommand(OutputCommands.LOCK__WIDGET, {
             isEnabled: widget => this.withWidget(widget, output => !output.isLocked),
             isVisible: widget => this.withWidget(widget, output => !output.isLocked),
-            execute: () => this.widget.then(widget => widget.lock())
+            execute: () => this.tryGetWidget()?.lock()
```

Comment:
here is `withWidget`, see https://github.com/eclipse-theia/theia/pull/8042/files#r443442525